interop/test_utils.go

```diff
@@ -454,6 +454,49 @@ func DoCancelAfterFirstResponse(tc testpb.TestServiceClient) {
 	}
 }
 
+// DoStatusCodeAndMessage checks that the status code is propagated back to the client.
+func DoStatusCodeAndMessage(tc testpb.TestServiceClient) {
+	var code int32 = 2
+	msg := "test status message"
+	respStatus := &testpb.EchoStatus{
+		Code:    &code,
+		Message: &msg,
+	}
+	// Test UnaryCall.
+	req := &testpb.SimpleRequest{
+		ResponseStatus: respStatus,
+	}
+	_, err := tc.UnaryCall(context.Background(), req)
+	if grpc.Code(err) != codes.Code(code) {
+		grpclog.Fatalf("/TestService/UnaryCall RPC compleled with error code %d, want %d", grpc.Code(err), code)
+	}
+	if err.Error() != fmt.Sprint("rpc error: code = ", code, " desc = ", msg) {
+		grpclog.Fatal("/TestService/UnaryCall unexpected RPC error message: ", err)
+	}
```

Comment:
I think the above can be better represented as:

expectedErr := grpc.Errorf(code, msg)
if _, err := tc.UnaryCall(...); err == nil || err.Error() != expectedErr.Error() {
   grpclog.Fatalf("%v.UnaryCall(_, _) = _, %v, want %v", err, expectedErr)
}